Libraries/CLI/src/cli.h

```diff
@@ -0,0 +1,155 @@
+//                      INCLUDES
+/* -------------------------------------------------- */
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include "board.h"
+#include "led.h"
+#include "mxc_device.h"
+#include "mxc_errors.h"
+#include "uart.h"
+#include "user-cli.h"
+
+/* -------------------------------------------------- */
+//                      MACROS
+/* -------------------------------------------------- */
+#define ENTER                   0X0D
+#define NEW_LINE                0x0A
+#define SPACE                   0x20
+#define TAB                     0x09
+#define BACKSPACE               0X08
+#define MAXBUFF                 2000
+#define DELETE                  0x7F
+#define DOLLAR                  0x24
+
+
+#define ARROW_KEY_CODE_1        0x1B
+#define ARROW_KEY_CODE_2        0x5B
+#define ARROW_KEY_CODE_LEFT     0x44
+#define ARROW_KEY_CODE_RIGHT    0x43
+#define ARROW_KEY_CODE_UP       0x41
+#define ARROW_KEY_CODE_DOWN     0x42
+
+
+/* -------------------------------------------------- */
+//                 FUNCTION PROTOTYPES
+/* -------------------------------------------------- */
+
+/**
+ * @brief Reads incoming bytes, Accumulate into line buffer, Echo's chars back to the other side and handles backspace.
+ * 		  It calls the process_command function upon pressing the ENTER key
+ *
+ * @param   user_char       User input of each character
+ */
+void line_accumlator(uint8_t user_char);
+
+/**
+ * @name process_command
+ *
+ * @brief Performs a Lexical analysis and tokenisis the user's commands
+ * 		  Lookup first token in a table of functions, dispatch to handler function
+ *
+ * @param   input   Character pointer containing the line accumulator input when enter key is pressed 
+ *
+ * @return  void
+ */
+void process_command(char *input);
+
+/** Prints out the user command prompt sequence o the uart console 
+*
+* @param void
+*
+* @return void
+*/
+void User_Prompt_Sequence(void);
+
+// Command table hander prototype with parameters
+typedef void (*command_handler_t)(int, char *argv[]);
+
+//Command table structure
+typedef struct
+{
+	const char *name;			/**< command string */
+	command_handler_t handler;	/**< function pointer of the handler function */
+	const char *help_string;	/**< help string of each command */
+} command_table_t;
+
+/*
+ * This table is an array of command_table_t structures that defines a set of supported commands in the program.
+ * Each command_table_t structure contains the name of the command, a function pointer to the corresponding command handler function,
+ * and a short description of what the command does.
+ *
+ * The structure of this lookup table makes it trivially easy to add a new command to this command processor.
+ */
+extern const command_table_t commands[];
+
+//Calculates the number of commands based on commands and the command table
+extern const int num_commands;
```

Comment:
Same here, inform the reader that they must provide the definition for this.  Since the `sizeof`-based definition we've implemented needs to be copy/pasted into the user's code, you can provide that to them in the comment as well.

Reply:
Shall add this under instructions to integrate CLI section of the markdown source file.